packages/core/src/components/forms/formGroup.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+import * as Classes from "../../common/classes";
+import { IIntentProps, IProps } from "../../common/props";
+
+export interface IFormGroupProps extends IIntentProps, IProps {
+    /**
+     * Whether form group should appear as non-interactive.
+     * Remember that input elements must be disabled separately.
+     */
+    disabled?: boolean;
+
+    /** Optional helper text. The given content will be wrapped in `.pt-form-helper-text` and displayed beneath `children`. */
+    helperText?: React.ReactNode;
+
+    /** Whether to render the label and children on a single line. */
+    inline?: boolean;
+
+    /** Label of this form group. */
+    label?: React.ReactNode;
+
+    /**
+     * `id` attribute of the labelable form element that this `FormGroup` controls,
+     * used as `<label for>` attribute.
+     */
+    labelFor?: string;
+
+    /**
+     * Whether this form input should appear as required (does not affect HTML form required status).
+     * Providing a boolean `true` value will render a default "required" message after the `label` prop.
+     * Providing a JSX value will render that content instead.
+     *
+     * _Note:_ the default message element is exposed as `FormGroup.DEFAULT_REQUIRED_CONTENT` and
+     * can be changed to provide a new global default for your app.
+     */
+    requiredLabel?: boolean | React.ReactNode;
```

Comment:
This doesn't appear as `boolean | React.ReactNode` in the docs. It appears as simply `React.ReactNode`. Was confusing at a glance. Anything we need to do to fix that?

![image](https://user-images.githubusercontent.com/443450/31466911-a829d3b2-ae8d-11e7-8913-3415b5447100.png)

Reply:
I believe that's because `ReactNode` is a union type that actually includes `boolean`.

Reply:
the docs should be enough. oh but I should add an `@default`